test_pull_requests

```diff
@@ -152,6 +185,115 @@ def submitted_tests_for_branch(branch)
   $submitted_tests[branch] ? $submitted_tests[branch] : $submitted_tests['*']
 end
 
+# Helper function to create appropriate regex for matching/extracting
+# :build_url field from bot comment
+def build_url_field_regex
+  if Properties &&
+    / \((#{Regexp::escape(Properties['jenkins_url'])}[^)]+)\)/
+  else
+    # Use this test to set the build url matching pattern
+    if ENV['BUILD_URL']
+      # Don't set build_url directly from the env var, since we want
+      # to make sure they match
+      / \((#{Regexp::escape(ENV['BUILD_URL'])})\)/
+    else
+      / \((https?:[^)]+)\)/
+    end
+  end
+end
+
+# Helper function to pull the test prefix from the settings object or
+# try to determine it from the provided bot_comment_body
+def extract_test_prefix(bot_comment_body, settings=nil)
+  if settings
+    settings['test_prefix']
+  else
+    # Do our best to find the prefix since we don't know it beforehand
+    if bot_comment_body =~ /^([^:(]+): /
+      $1.strip
+    elsif bot_comment_body =~ /^([^ (]+) /
+      $1.strip
+    end
+  end
+end
+
+# Pulls out as much metadata from the bot comment as it can, including
+# state (AKA content), Jenkins build URL, base repo commit ID, and the
+# extended tests metadata
+def extract_bot_comment_fields(bot_comment_body, settings=nil)
```

Comment:
* adding a field is more work for the caller
* `test_prefix` and `test_name` are tightly coupled values, so it is reasonable to pass them in some kind of data structure
* if we want this function to operate on another value from `settings` in the future, we would have to add another field if we go with your suggestion
* the exception-generating conditional block would be less clear if we had to check `test_name.nil?` instead of `settings.nil?`